### PR TITLE
Fix #140: On enums and scope.

### LINKED ISSUE #140: On enums and scope
With the libdvbv5 C header file transform, a lot of named enums are created. This is fine per se. So for example:
```
typedef enum fe_delivery_system {
	SYS_UNDEFINED,
	SYS_DVBC_ANNEX_A,
	SYS_DVBC_ANNEX_B,
	SYS_DVBT,
	SYS_DSS,
	SYS_DVBS,
	SYS_DVBS2,
	SYS_DVBH,
	SYS_ISDBT,
	SYS_ISDBS,
	SYS_ISDBC,
	SYS_ATSC,
	SYS_ATSCMH,
	SYS_DTMB,
	SYS_CMMB,
	SYS_DAB,
	SYS_DVBT2,
	SYS_TURBO,
	SYS_DVBC_ANNEX_C,
} fe_delivery_system_t;
```
is transformed to:
```
enum fe_delivery_system
{
    SYS_UNDEFINED = 0,
    SYS_DVBC_ANNEX_A = 1,
    SYS_DVBC_ANNEX_B = 2,
    SYS_DVBT = 3,
    SYS_DSS = 4,
    SYS_DVBS = 5,
    SYS_DVBS2 = 6,
    SYS_DVBH = 7,
    SYS_ISDBT = 8,
    SYS_ISDBS = 9,
    SYS_ISDBC = 10,
    SYS_ATSC = 11,
    SYS_ATSCMH = 12,
    SYS_DTMB = 13,
    SYS_CMMB = 14,
    SYS_DAB = 15,
    SYS_DVBT2 = 16,
    SYS_TURBO = 17,
    SYS_DVBC_ANNEX_C = 18
}

alias fe_delivery_system_t = fe_delivery_system;
```
However, the line:
```
#define SYS_DVBC_ANNEX_AC	SYS_DVBC_ANNEX_A
```
gets transformed into:
```
enum SYS_DVBC_ANNEX_AC = SYS_DVBC_ANNEX_A;
```
where it needs to be transformed to:
```
enum SYS_DVBC_ANNEX_AC = fe_delivery_system.SYS_DVBC_ANNEX_A;
```

### PR DESCRIPTION
**Pull request by ciechowoj**
As is the title.

### CODE REVIEW

File: dstep/Configuration.d [hunk 1]
```diff
@@ -53,6 +53,10 @@ struct Configuration
     @("reduce-aliases", "Reduce primitive type aliases [default].")
     bool reduceAliases = true;
 
+    /// generate aliases for enum items in global scope
+    @("alias-enum-items", "Generate aliases for enum items in global scope [default].")
```
Comment (jacob-carlborg) [inline dstep/Configuration.d]: `items` -> `members`.

Comment (ciechowoj) [inline dstep/Configuration.d]: Renamed.

File: unit_tests/EnumUnitTests.d [hunk 2]
```diff
@@ -216,6 +216,38 @@ D");
 
 }
 
+// Test aliasing of all enum items.
```
Comment (jacob-carlborg) [inline unit_tests/EnumUnitTests.d]: Here as well.

Comment (ciechowoj) [inline unit_tests/EnumUnitTests.d]: Sigh...
Extended the commit message as well.

**Comment by jacob-carlborg** [comment]
👌.

**Comment by jacob-carlborg** [comment]
Could you please create a PR that updates the changelog with the PRs I just merged. I merged them a bit too quickly 😃.

**Comment by ciechowoj** [comment]
Sure.

**Comment by jacob-carlborg** [comment]
I just noticed that the following code:

```c
struct Foo
{
    enum Bar
    {
        asd
    };
};
```

Incorrectly translates, with the new `--alias-enum-members` flag, to:

```d
extern (C):

struct Foo
{
    enum Bar
    {
        asd = 0
    }
}

alias asd = Bar.asd;
```

The alias should be placed in the parent scope. Or perhaps that's not valid C code?

**Comment by jacob-carlborg** [comment]
Here's a better example:

```c
struct Foo
{
    enum Bar
    {
        asd
    } bar;
};
```

**Comment by ciechowoj** [comment]
The alias should be placed in the parent scope not the global scope.

No, in C enums are exported to global scope.

e.g.

```
#include <stdio.h>

struct Foo
{
    enum Bar
    {
        asd = 1
    };
};


int main() {
	printf("%d", asd);
	return 0;
}
```
This compiles fine and prints 1.

Anyway it should be `alias asd = Foo.Bar.asd;`

**Comment by jacob-carlborg** [comment]
Ah, I was running the C++ compiler, not the C compiler. My bad.

**Pull request merged**